cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/sqlite/SQLDatabaseFactory.java

```diff
@@ -48,12 +80,25 @@
      * @throws SQLException if the database cannot be opened.
      */
     public static SQLDatabase createSQLDatabase(String dbFilename, KeyProvider provider) throws IOException, SQLException {
+        makeSureFileExists(dbFilename);
+        return internalCreateSQLDatabase(dbFilename, provider);
+    }
+
+    /**
+     * Internal method for creating a SQLDatabase that allows a null filename to create an in-memory
+     * database which can be useful for performing checks, but creating in-memory databases is not
```

Comment:
Added comment in b19786cd545eac2208a498ee5df58ff9816414c6. I think that ought to be sufficient, but it it isn't we could always add a check to `internalCreateSQLDatabase` like:
```java
if (dbFilename == null && !(provider instanceof NullKeyProvider)) {
     throw new IllegalArgumentException("In-memory datastore must use NullKeyProvider.");
 }
```